darts/models/forecasting/forecasting_model.py

```diff
@@ -920,6 +944,10 @@ def retrain_func(
         for idx, series_ in enumerate(outer_iterator):
             past_covariates_ = past_covariates[idx] if past_covariates else None
             future_covariates_ = future_covariates[idx] if future_covariates else None
+            if isinstance(sample_weight, str):
+                sample_weight_ = sample_weight
+            else:
+                sample_weight_ = sample_weight[idx] if sample_weight else None
```

Comment:
The method `_historical_forecasts_sanity_checks` should probably be updated to check that the `sample_weight` parameter is valid